sci-geosciences/mapserver/mapserver-7.0.7.ebuild

```diff
@@ -0,0 +1,220 @@
+)
+
+S=${WORKDIR}/${MY_P}
+
+pkg_setup() {
+	if use apache ; then
+		webapp_pkg_setup
+	fi
+	use perl && perl_set_version
+	#use ruby && ruby-ng_pkg_setup
+}
+
+src_unpack() {
+	default
+	# HACK: Make symlinks for php targets
+	local slot
+	for slot in $(php_get_slots); do
+		ln -s "${PHP_EXT_S}" "${WORKDIR}/${slot}" || die
+	done
+}
+
+src_prepare() {
+	local glibdir="${EPREFIX}/usr/include/glib-2.0"
+	local usrglibdir="${EPREFIX}/usr/$(get_libdir)/glib-2.0/include"
+
+	sed -e "s:^include_directories(:&${glibdir})\ninclude_directories(:" \
+		-i "${S}/CMakeLists.txt" || die
+	sed -e "s:include_directories(:&${usrglibdir})\ninclude_directories(:" \
+		-i "${S}/CMakeLists.txt" || die
+
+	# For enable compilation with proj-6
+	if use proj ; then
+		append-cppflags -DACCEPT_USE_OF_DEPRECATED_PROJ_API_H=1
+	fi
+
+	cmake-utils_src_prepare
+}
+
+src_configure() {
+	local mycmakeargs=(
+		"-DCMAKE_SKIP_RPATH=ON"
+		"-DWITH_SDE=OFF"
+		"-DWITH_ICONV=ON"
+		"-DWITH_GD=ON"
+		"-DWITH_GIF=ON"
+		"-DWITH_CURL=ON"
+		"-DINSTALL_LIB_DIR=${ROOT}usr/$(get_libdir)"
+		"-DWITH_APACHE_MODULE=$(usex apache ON OFF)"
+		"-DWITH_PROJ=$(usex proj ON OFF)"
+		"-DWITH_WMS=$(usex proj ON OFF)"
+		"-DWITH_KML=$(usex xml ON OFF)"
+		"-DWITH_FCGI=$(usex apache ON OFF)"
+		"-DWITH_GEOS=$(usex geos ON OFF)"
+		"-DWITH_GDAL=$(usex gdal ON OFF)"
+		"-DWITH_OGR=$(usex gdal ON OFF)"
+		"-DWITH_ORACLESPATIAL=$(usex oracle ON OFF)"
+		"-DWITH_POSTGIS=$(usex postgis ON OFF)"
+		"-DWITH_MYSQL=$(usex mysql ON OFF)"
+		"-DWITH_LIBXML2=$(usex xml ON OFF)"
+		"-DWITH_XMLMAPFILE=$(usex xslt ON OFF)"
+		"-DWITH_FRIBIDI=$(usex bidi ON OFF)"
+		"-DWITH_HARFBUZZ=$(usex bidi ON OFF)"
+		"-DWITH_CAIRO=$(usex cairo ON OFF)"
+		"-DWITH_PHP=$(usex php ON OFF)"
+		"-DWITH_PYTHON=$(usex python ON OFF)"
+		"-DWITH_PERL=$(usex perl ON OFF)"
+	)
+
+	if use gdal && use proj ; then
+		mycmakeargs+=( "-DWITH_WFS=ON"
+				"-DWITH_WCS=ON"
+				"-DWITH_CLIENT_WMS=ON"
+				"-DWITH_CLIENT_WFS=ON"
+				"-DWITH_SOS=$(usex xml ON OFF)"
+			)
+	else
+		mycmakeargs+=( "-DWITH_WFS=OFF"
+			"-DWITH_WCS=OFF"
+			"-DWITH_CLIENT_WMS=OFF"
+			"-DWITH_CLIENT_WFS=OFF"
+			"-DWITH_SOS=OFF"
+		)
+	fi
+
+	if use php ; then
+		local slot
+		for slot in $(php_get_slots); do
+			local php_config="${EPREFIX}/usr/$(get_libdir)/${slot}/bin/php-config"
+			[[ -x ${php_config} ]] \
+				|| die "php-config '${php_config}' not valid or not executable"
+
+			local php_include_dir=$(${php_config} --include-dir)
+			[[ -d ${php_include_dir} ]] || die "PHP Include dir not found or not valid"
+
+			mycmakeargs+=(
+				-DPHP5_CONFIG_EXECUTABLE="${php_config}"
+				-DPHP5_INCLUDES="${php_include_dir}"
+			)
+		done
+	fi
+
+	cmake-utils_src_configure
+}
+
+src_compile() {
+	cmake-utils_src_compile
+}
+
+src_install() {
+	cmake-utils_src_install
+
+	if use php ; then
+		php-ext-source-r3_createinifiles
+	fi
+
+	if use apache ; then
+		webapp_src_preinst
+	fi
+
+	exeinto "${MY_CGIBINDIR}"
+	doexe "${S}_build/mapserv"
```

Comment:
not needed

Reply:
Hi,
WITH_GD support was removed and WITH_SDE is no longer supported from mapserver-7.x.x (see [MapServer 6.4 to 7.0 Migration](https://mapserver.org/MIGRATION_GUIDE.html#mapserver-6-4-to-7-0-migration))